packages/shared/lib/services/environment.service.ts

```diff
@@ -547,4 +542,12 @@ class EnvironmentService {
     }
 }
 
+export async function hashSecretKey(key: string) {
+    if (!ENCRYPTION_KEY) {
+        return key;
+    }
+
+    return (await pbkdf2(key, ENCRYPTION_KEY, 310000, 32, 'sha256')).toString('base64');
```

Comment:
curious to know how you picked the 31000 iterations and 32 key length value?

Reply:
copied from our password hash function